src/types.ts

```diff
@@ -31,6 +31,6 @@ export type IconComponent = FunctionComponent<JSX.SVGAttributes<SVGSVGElement>>;
  * animate the mounting and unmounting of a child component.
  */
 export type TransitionComponent = FunctionComponent<{
-  visible: boolean;
+  direction?: 'in' | 'out';
```

Comment:
Does this mean that all `TransitionComponent`s have to support both an `in` and an `out` transition? Is this an OK assumption to make?